python312Packages.py-slvs: init at 1.0.6

traverseda opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

As per #303561

Makes it much easier to use blender's cad-sketcher extension.

` blenderWithPySlvs = blender.withPackages (p: [py-slvs]);`

Closes #303561

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

traverseda commented:
@ofborg python3Packages.py-slvs

TomaSajt commented on pkgs/development/python-modules/py-slvs/default.nix:
```diff
@@ -0,0 +1,37 @@
+{ lib, blender, python3Packages, pkgs, fetchurl }:
```
```suggestion
{
  lib,
  buildPythonPackage,
  fetchPypi,
  swig,
  cmake,
  ninja,
  setuptools,
  scikit-build,
}:
```

Since this file if being called with `callPackage` from inside `pkgs/top-level/python-packages.nix` this means that both python and non-python packages will be populated automatically in this top attrset.

By doing this you can remove the explicit references to `python3Packages` and `pkgs`.

TomaSajt commented on pkgs/development/python-modules/py-slvs/default.nix:
```diff
@@ -0,0 +1,37 @@
+
+  src = python3Packages.fetchPypi {
+    pname = "py_slvs";
+    version = "1.0.6";
```
If you're using `rec` then you can inherit the version value from the outer attrset, making the package less work to update (only 1 place to change).
```suggestion
    inherit version;
```

TomaSajt commented on pkgs/development/python-modules/py-slvs/default.nix:
```diff
@@ -0,0 +1,37 @@
+    swig
+  ];
+
+  propagatedBuildInputs = with python3Packages; [
```
I'd recommend changing over from `propagatedBuildInputs` to the new naming for build-time python dependencies: `build-system`.

If you're wondering: this naming is taken directly from `pyproject.toml`, which also uses `build-system`

```suggestion
  build-system = [
```

TomaSajt commented on pkgs/development/python-modules/py-slvs/default.nix:
```diff
@@ -0,0 +1,37 @@
+
+  dontUseCmakeConfigure = true;
+
+  meta = with pkgs.lib; {
```
```suggestion
  meta = with lib; {
```

traverseda commented:
Alright, I'll figure out the new issues in the morning, thanks

eclairevoyant commented on pkgs/development/python-modules/py-slvs/default.nix:
```diff
@@ -0,0 +1,46 @@
+  scikit-build,
+}:
+
+python3Packages.buildPythonPackage rec {
```
```suggestion
buildPythonPackage rec {
```

eclairevoyant commented on pkgs/development/python-modules/py-slvs/default.nix:
```diff
@@ -0,0 +1,46 @@
+  pname = "py-slvs";
+  version = "1.0.6";
+
+  src = python3Packages.fetchPypi {
```
```suggestion
  src = fetchPypi {
```

eclairevoyant commented on pkgs/development/python-modules/py-slvs/default.nix:
```diff
@@ -0,0 +1,46 @@
+
+  dontUseCmakeConfigure = true;
+
+  meta = with lib; {
```
```suggestion
  meta = {
```

eclairevoyant commented on pkgs/development/python-modules/py-slvs/default.nix:
```diff
@@ -0,0 +1,46 @@
+  meta = with lib; {
+    description = "Python binding of SOLVESPACE geometry constraint solver";
+    homepage = "https://github.com/realthunder/slvs_py";
+    license = licenses.gpl3;
```
```suggestion
    license = lib.licenses.gpl3;
```

also please use gpl3Only or gpl3Plus in accordance with upstream

eclairevoyant commented on pkgs/development/python-modules/py-slvs/default.nix:
```diff
@@ -0,0 +1,46 @@
+    description = "Python binding of SOLVESPACE geometry constraint solver";
+    homepage = "https://github.com/realthunder/slvs_py";
+    license = licenses.gpl3;
+    maintainers = with maintainers; [
```
```suggestion
    maintainers = with lib.maintainers; [
```

eclairevoyant commented on pkgs/development/python-modules/py-slvs/default.nix:
```diff
@@ -0,0 +1,46 @@
+
+  dontUseCmakeConfigure = true;
+
+  meta = with lib; {
```
we should probably have an imports check as well

traverseda replied:
Just check that it's importable, not run full test cases, right?

eclairevoyant replied:
No, I'm referring to https://nixos.org/manual/nixpkgs/unstable/#using-pythonimportscheck here

eclairevoyant reviewed (dismissed):
Thanks for the contribution! Some feedback below.

Also please rebase into 2 commits:
* `maintainers: add traverseda`
* `python312Packages.py-slvs: init at 1.0.6`

traverseda commented on pkgs/development/python-modules/py-slvs/default.nix:
```diff
@@ -0,0 +1,47 @@
+  ninja,
+  setuptools,
+  scikit-build,
+  maintainers,
```
Yeah, I think I've got that figured out. I'll be going through your feedback in more detail once I got off work. I didn't see your feedback before the most recent changes I pushed, and it won't let me merge them in automatically, but I'll be going over it all.

traverseda commented:
I hate to ask this here, but is there an easier way for me to test this than doing a full nixos-rebuild-switch and modifying the flake I use to build my system? I'm having a hard time figuring that out

TomaSajt commented:
If you go into your local nixpkgs directory, you can run `nix build .#python3Packages.py-slvs` or if you want to use the older cli tools, then `nix-build -A python3Packages.py-slvs`

traverseda commented:
Alright, next step for me will be rebasing it I believe, I *think* I've got all the feedback resolved, although I messed up the workflows a little bit on my end.

traverseda committed:
maintainers: add traverseda

traverseda committed:
python312Packages.py-slvs: init at 1.0.6

Forgot a typo

Apply suggestions from code review

Co-authored-by: Toma <62384384+TomaSajt@users.noreply.github.com>

Explicit imports

Process feedback from eclairevoyant

traverseda commented:
I've got this rebased, everything should be ready to go

traverseda commented:
Just a reminder that this is ready to go, to the best of my knowledge. Anything I can do to move this along?
